services/conversation/v1-exp.js

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function (RED) {
+  var cfenv = require('cfenv'), watson = require('watson-developer-cloud'), service = null,
+    sUsername = null, sPassword = null;
+  
+  service = cfenv.getAppEnv().getServiceCreds(/conversation/i);
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+  RED.httpAdmin.get('/watson-conversation/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+ 
+  
+  function verifyPayload(node, msg) {
+    if (!msg.payload) {
+      this.status({fill:'red', shape:'ring', text:'missing payload'}); 
+      node.error('Missing property: msg.payload', msg);
+      return false;
+    }
+    return true;
+  }
+
+  function verifyInputs(node, msg, config) {
+    // workspaceid can be either configured in the node, 
+    // or sent into msg.params.workspace_id
+    if (!config.workspaceid || '' !== config.workspaceid) {
+      node.workspaceid = config.workspaceid;
+      return true;
+    } else if (!msg.params || '' !== msg.params.workspace_id) {
+      node.workspaceid = msg.params.workspace_id;
+      return true;
+    }
+    node.error('Missing workspace_id. check node documentation.',msg);
+    return false;
+  }
+
+  function verifyServiceCredentials(node, msg) {
+    // If it is present the newly provided user entered key 
+    // takes precedence over the existing one. 
+    //node.apikey = sAPIKey || node.credentials.apikey;
+    var userName = sUsername || node.credentials.username,
+      passWord = sPassword || node.credentials.password;
+    
+    if (!userName || !passWord) {
+      node.status({fill:'red', shape:'ring', text:'missing credentials'});          
+      node.error('Missing Watson Conversation API service credentials', msg);
+      return false;
+    }
+    node.service = watson.conversation({
+      username: userName,
+      password: passWord,
+      version_date: '2016-05-19',
+      version: 'v1-experimental'
+    });
+    return true;
+  }
+
+  function processResponse(err, body, node, msg) {
+    if (err != null && body == null) {
+      node.status({fill:'red', shape:'ring', 
+        text:'call to watson conversation service failed'}); 
+      msg.result = {};
+      if (err.code == null) {
+        msg.result['error']=err;
+      } else {
+        msg.result['error_code'] = err.code;
+        if (!err.error) {
+          msg.result['error'] = err.error;
+        }        
+      }
+      node.error(err);
+      return;
+    }
+    msg.result = body;
+    msg.payload = 'see msg.result';
+    node.send(msg); 
+    node.status({});
+  }
+
+  function execute(params, node, msg) {
+    node.status({fill:'blue', shape:'dot' , text:'Calling Conversation service ...'});
+    params.workspace_id=node.workspaceid;
+    params.input={text:msg.payload};
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2777516067/issues/source?bid=3225115&fileBranchId=3364374#l102)